easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -702,10 +702,18 @@ def parse(self):
             # parse dependency specifications
             # it's important that templating is still disabled at this stage!
             self.log.info("Parsing dependency specifications...")
-            self['dependencies'] = [self._parse_dependency(dep) for dep in self['dependencies']]
-            self['hiddendependencies'] = [
-                self._parse_dependency(dep, hidden=True) for dep in self['hiddendependencies']
-            ]
+
+            def remove_false_versions(deps):
+                ret = []
+                for dep in deps:
+                    if isinstance(dep, dict) and dep['version'] is False:
+                        continue
+                    ret.append(dep)
+                return ret
```

Comment:
How about a (faster) list comprehension:
```suggestion
                return [dep for dep in deps if not (isinstance(dep, dict) and dep['version'] is False)]
```